### PR TITLE
Update _index.md

### PR DESCRIPTION
**Pull request by justinegeffen**
Typo and rewording.

@aaronrothschild, wondering if we could add something for the community with a link to help wanted tickets for other languages?

### CODE REVIEW

**Review by hanzei — APPROVED** [review]
LGTM

**Comment by aaronrothschild** [comment]
I think it would be great if we could ask for explicit "equivalents" of the Go documentation pages we have now.  ie: quickstart (I think we have a JS version of that?):
- HelloWorld anatomy in JS and Python
- Hello Oauth2 anatomy in JS and Python
- ...? 

@hanzei, I'm assuming the priority is somewhat: Go, JS, Python, .... ? I'd love to base that on data about which languages are most popular amongst our developers in the community....would Orbit help answer that?

**Comment by hanzei** [comment]
> I think it would be great if we could ask for explicit "equivalents" of the Go documentation pages we have now. ie: quickstart (I think we have a JS version of that?):
> 
> * HelloWorld anatomy in JS and Python
> * Hello Oauth2 anatomy in JS and Python
> * ...?
> 
> @hanzei, I'm assuming the priority is somewhat: Go, JS, Python, .... ? I'd love to base that on data about which languages are most popular amongst our developers in the community....would Orbit help answer that?

To me the only priority is JS. The rest can follow later (or never). Maybe Java also.

It's really a question about which languages are most popular among the possible _app_ developers. Orbit doesn't cover that (It's still broken BTW). Do you have insights from customers?

**Comment by justinegeffen** [comment]
@aaronrothschild, I've added a new blurb to the landing page. PLMK what you think and/or edit as needed. :)

File: site/content/integrate/getting-started/how-should-i-integrate.md [hunk 1]
```diff
@@ -27,6 +27,10 @@ That's not enough? You can include <a target="_blank" href="https://docs.matterm
 
 To build a richer bot integration, you can make full use of the [Mattermost REST API](/integrate/rest-api/). Everything that you see a Mattermost client doing, your integration can do too with this API.
 
+### I want to develop a cloud-based app
```
Comment (hanzei) [inline site/content/integrate/getting-started/how-should-i-integrate.md:30]: Cloud based?

Comment (justinegeffen) [reply site/content/integrate/getting-started/how-should-i-integrate.md:30]: I couldn't think of a better way to frame it. How about "I want a more flexible option"?

Comment (hanzei) [reply site/content/integrate/getting-started/how-should-i-integrate.md:30]: Isn't the selling point the rich UX possibilities?

Comment (justinegeffen) [inline site/content/integrate/getting-started/how-should-i-integrate.md:30]:
```suggestion
### I want to build something with more interactivity
```

File: site/content/integrate/apps/_index.md [hunk 2]
```diff
@@ -29,9 +29,20 @@ Apps are available as a Developers Preview and we're looking for your feedback!
 
 Read the [quick start guide]({{< ref  "quick-start-go" >}}) to learn how to write your first app.
 
-
 ## FAQ
 
+### When would you build an app vs a custom slash command vs a webhook vs a plugin?
+
+That depends on your use case, as they each have benefits. Apps can be advertised, shared, and installed by others via the Marketplace and users can get started with a slash command. Comparatively, a custom slash and webhooks command requires a lot of configuration, is very rigid - and is limited to your own server.
```
Comment (hanzei) [inline site/content/integrate/apps/_index.md:36]: > Apps can be advertised, shared, and installed by others via the Marketplace 

Not really right now. It's unclear how the Marketplace will allow installation for Apps besides the Mattermost build ones.

File: site/content/integrate/apps/bindings/_index.md [hunk 3]
```diff
@@ -5,21 +5,23 @@ description: "TODO"
 weight: 50
 ---
 
-Bindings ([godoc](https://pkg.go.dev/github.com/mattermost/mattermost-plugin-apps/apps#Binding)) are what establish the relationship between locations and calls. Whenever it's called, you have to provide the list of bindings available according to the context. The context for the bindings call includes the app ID, bot access token, team ID, channel ID, acting user ID and Mattermost site URL. By default it doesn't expand any value.
+Bindings ([godoc](https://pkg.go.dev/github.com/mattermost/mattermost-plugin-apps/apps#Binding)) are what establish the relationship between locations and calls. Whenever it's called, you have to provide the list of bindings available according to the context. The context for the bindings call includes the app ID, bot access token, team ID, channel ID, acting user ID, and Mattermost site URL. By default it doesn't expand any value.
+
+**Note:** When the available bindings are changed, the user must change the channel in order to see the updated bindings.
```
Comment (hanzei) [inline site/content/integrate/apps/bindings/_index.md:10]: The wording is a bit misleading. The point is about the per-user bindings, not the computation in general.
```suggestion
**Note:** When the available bindings for a given user are changed, the user must change the channel in order to see the updated bindings.
```

File: site/content/integrate/getting-started/how-should-i-integrate.md [hunk 4]
```diff
@@ -26,6 +26,10 @@ That's not enough? You can include <a target="_blank" href="https://docs.matterm
 
 To build a richer bot integration, you can make full use of the [Mattermost REST API](/integrate/rest-api/). Everything that you see a Mattermost client doing, your integration can do too with this API.
 
+### I want to develop a cloud-based app 
+
+That's no problem. Our apps framework supports apps written in any language that supports HTTP, and work on our Mobile and Desktop clients. Apps can be hosted anywhere, on a Digital Ocean droplet, AWS EC2, or a server in your data centre. Take a look at our [Quickstart Guide](https://developers.mattermost.com/integrate/apps/quick-start-go/).
```
Comment (hanzei) [inline site/content/integrate/getting-started/how-should-i-integrate.md:31]: Slash command and webhooks also use HTTP as the transport protocol and can to hosted anywhere. It point is that Apps provide a _framework_ with utils like a KV store, bot account creation  and OAuth permission scopes and -most notable- rich UX entry points.

File: site/content/integrate/getting-started/how-should-i-integrate.md (hunk 1 above)
Comment (hanzei) [reply site/content/integrate/getting-started/how-should-i-integrate.md:29]: Apply this suggestion?

File: site/content/integrate/apps/_index.md (hunk 2 above)
Comment (hanzei) [inline site/content/integrate/apps/_index.md:36]: What does "and is limited to your own server" mean?

Comment (hanzei) [reply site/content/integrate/apps/_index.md:36]: Please also see https://github.com/mattermost/mattermost-developer-documentation/pull/804#discussion_r622019565 on the selling point of apps.

Comment (justinegeffen) [inline site/content/integrate/apps/_index.md:36]:
```suggestion
That depends on your use case, as they each have benefits. Apps can be advertised, shared, and installed by others via the Marketplace and users can get started with a slash command. Comparatively, a custom slash and webhooks command requires a lot of configuration and is less interactive.
```

File: site/content/integrate/getting-started/how-should-i-integrate.md [hunk 5]
```diff
@@ -26,6 +26,10 @@ That's not enough? You can include <a target="_blank" href="https://docs.matterm
 
 To build a richer bot integration, you can make full use of the [Mattermost REST API](/integrate/rest-api/). Everything that you see a Mattermost client doing, your integration can do too with this API.
 
+### I want to build an interactive app
+
+That's no problem. Our apps framework supports apps written in any language that supports HTTP, and work on our Mobile and Desktop clients. Apps can be hosted anywhere, on a Digital Ocean droplet, AWS EC2, or a server in your data centre. Take a look at our [Quickstart Guide](https://developers.mattermost.com/integrate/apps/quick-start-go/).
```
Comment (justinegeffen) [inline site/content/integrate/getting-started/how-should-i-integrate.md:31]:
```suggestion
That's no problem. Our Apps provide a framework with utilities like a KV store, bot account creation, OAuth permission scopes, and rich UX entry points.

Apps can be written in any language that supports HTTP, and work on our Mobile and Desktop clients. Apps can be hosted anywhere, on a Digital Ocean droplet, AWS EC2, or a server in your data centre. Take a look at our [Quickstart Guide](https://developers.mattermost.com/integrate/apps/quick-start-go/).
```

**Comment by justinegeffen** [comment]
@aaronrothschild, ping on this. :)

**Comment by justinegeffen** [comment]
@mickmister, adding you to this - I'd like your feedback on the wording I've used. :)

File: site/content/integrate/apps/_index.md [hunk 6]
```diff
@@ -31,6 +31,18 @@ Read the [JavaScript start guide]({{< ref "quick-start-js" >}}) or [Go quick sta
 
 ## FAQ
 
+### When would you build an app vs a custom slash command vs a webhook vs a plugin?
+
+That depends on your use case, as they each have benefits. Apps can be advertised, shared, and installed by others via the Marketplace and users can get started with a slash command. Comparatively, a custom slash and webhooks command requires a lot of configuration and is less interactive.
```
Comment (aaronrothschild) [inline site/content/integrate/apps/_index.md:36]:
```suggestion
That depends on your use case, as they each have benefits. Apps can be advertised, shared, and installed by others via the Marketplace. Comparatively, a custom slash command requires some configuration and is less interactive.

The built-in incoming webhook is rigid and requires specific fields to be posted to it's URL, but some external systems don't allow the customization of the fields included in the webhook payload.   
Compared to built-in webhooks, An App can be customized to receive an HTTP webhook posting from another system, and can process the incoming data then augment it or make an actionable message in a channel.  

A plugin should be used when you need to alter the React UX in Mattermost or have an application that requires low-latency with the server (such as replacing characters in any message). Currently, plugins have several UX hooks that Apps cannot access, however we plan to add/migrate more UX hooks into the Apps Framework over time. 
```

File: site/content/integrate/apps/_index.md [hunk 7]
```diff
@@ -31,6 +31,18 @@ Read the [JavaScript start guide]({{< ref "quick-start-js" >}}) or [Go quick sta
...
+The App framework provides a few differences from plugins.
+
+- We’ve made our interactive elements easier to use and develop.
+- Works on both desktop and mobile clients without any extra code.
+- You host the app code or build for Matterhost and distribute through the Marketplace.
+
 ### What language should I use to write apps?
 
-Any language you want. The only langue with an official driver is go right now.
+Any language you want. However, the only language with an official driver right now is Go.
```
Comment (aaronrothschild) [inline site/content/integrate/apps/_index.md:48]:
```suggestion
Any language you want. However, the only language with an official driver right now is Go.  we are planning to release other drivers in the future in conjunction with our community.
```

File: site/content/integrate/apps/bindings/_index.md [hunk 8]
```diff
@@ -5,21 +5,23 @@ description: "TODO"
 weight: 50
 ---
 
-Bindings ([godoc](https://pkg.go.dev/github.com/mattermost/mattermost-plugin-apps/apps#Binding)) are what establish the relationship between locations and calls. Whenever it's called, you have to provide the list of bindings available according to the context. The context for the bindings call includes the app ID, bot access token, team ID, channel ID, acting user ID and Mattermost site URL. By default it doesn't expand any value.
+Bindings ([godoc](https://pkg.go.dev/github.com/mattermost/mattermost-plugin-apps/apps#Binding)) are what establish the relationship between locations and calls. Whenever it's called, you have to provide the list of bindings available according to the context. The context for the bindings call includes the app ID, bot access token, team ID, channel ID, acting user ID, and Mattermost site URL. By default it doesn't expand any value.
+
+**Note:** When the available bindings for a given user are changed, the user must change the channel in order to see the updated bindings.
```
Comment (aaronrothschild) [inline site/content/integrate/apps/bindings/_index.md:10]:
```suggestion
**Note:** When the available bindings for a given user are changed, the user must change the channel or refresh the page in order to see the updated bindings.
```

**Review by aaronrothschild — APPROVED** [review]

File: site/content/integrate/apps/_index.md [hunk 9]
```diff
@@ -31,6 +31,22 @@ Read the [JavaScript start guide]({{< ref "quick-start-js" >}}) or [Go quick sta
...
+
+A plugin should be used when you need to alter the React UX in Mattermost or have an application that requires low-latency with the server (such as replacing characters in any message). Currently, plugins have several UX hooks that Apps cannot access, however we plan to add/migrate more UX hooks into the Apps Framework over time.
+
+### What's the difference between the apps framework and the plugin framework?
+
+The App framework provides a few differences from plugins.
+
+- We’ve made our interactive elements easier to use and develop.
+- Works on both desktop and mobile clients without any extra code.
+- You host the app code or build for Matterhost and distribute through the Marketplace.
```
Comment (mickmister) [inline site/content/integrate/apps/_index.md:48]: Is Matterhost a real thing?

File: site/content/integrate/apps/_index.md [hunk 10]
```diff
@@ -31,6 +31,22 @@ Read the [JavaScript start guide]({{< ref "quick-start-js" >}}) or [Go quick sta
...
+
 ### What language should I use to write apps?
 
-Any language you want. The only langue with an official driver is go right now.
+Any language you want. However, the only language with an official driver right now is Go.  we are planning to release other drivers in the future in conjunction with our community.
```
Comment (mickmister) [inline site/content/integrate/apps/_index.md:52]:
```suggestion
Any language you want. We currently have an official driver for Go. We are planning to release other drivers in the future in conjunction with our community.
```

File: site/content/integrate/getting-started/how-should-i-integrate.md [hunk 11]
```diff
@@ -26,6 +26,12 @@ That's not enough? You can include <a target="_blank" href="https://docs.matterm
 
 To build a richer bot integration, you can make full use of the [Mattermost REST API](/integrate/rest-api/). Everything that you see a Mattermost client doing, your integration can do too with this API.
 
+### I want to build an interactive app
+
+That's no problem. Our Apps provide a framework with utilities like a KV store, bot account creation, OAuth permission scopes, and rich UX entry points.
```
Comment (mickmister) [inline site/content/integrate/getting-started/how-should-i-integrate.md:31]:
```suggestion
That's no problem. Our Apps provide a framework with utilities like a key-value store, bot account creation, OAuth permission scopes, and rich UX entry points.
```

File: site/content/integrate/apps/bindings/_index.md [hunk 12]
```diff
@@ -5,21 +5,23 @@ description: "TODO"
 weight: 50
 ---
 
-Bindings ([godoc](https://pkg.go.dev/github.com/mattermost/mattermost-plugin-apps/apps#Binding)) are what establish the relationship between locations and calls. Whenever it's called, you have to provide the list of bindings available according to the context. The context for the bindings call includes the app ID, bot access token, team ID, channel ID, acting user ID and Mattermost site URL. By default it doesn't expand any value.
+Bindings ([godoc](https://pkg.go.dev/github.com/mattermost/mattermost-plugin-apps/apps#Binding)) are what establish the relationship between locations and calls. Whenever it's called, you have to provide the list of bindings available according to the context. The context for the bindings call includes the app ID, bot access token, team ID, channel ID, acting user ID, and Mattermost site URL. By default it doesn't expand any value.
```
Comment (mickmister) [inline site/content/integrate/apps/bindings/_index.md:8]:
```suggestion
Bindings ([godoc](https://pkg.go.dev/github.com/mattermost/mattermost-plugin-apps/apps#Binding)) are what establish the relationship between locations and calls. Whenever the bindings route is called, your app needs to provide the list of bindings available according to the context ([godoc](https://pkg.go.dev/github.com/mattermost/mattermost-plugin-apps/apps#Context)). Some fields included in the context:
- Your app's bot user access token
- The Mattermost Site URL
- The id of the user requesting the bindings (acting user id)
- The id of the team the user is currently focused on
- The id of the channel the user is currently focused on
- The id of the post the user is currently focused on (if applicable)
```

> By default it doesn't expand any value.

I'm not sure if we explain `expand` anywhere. Saying "By default it doesn't expand any value." doesn't mean anything if `expand` isn't itself explained

File: site/content/integrate/apps/bindings/_index.md [hunk 13]
```diff
@@ -5,21 +5,23 @@ description: "TODO"
...
+
+**Note:** When the available bindings for a given user are changed, the user must change the channel or refresh the page in order to see the updated bindings.
```
Comment (mickmister) [inline site/content/integrate/apps/bindings/_index.md:10]:
```suggestion
**Note:** When an OAuth2 process is completed, the client's bindings are automatically refreshed. For any other case where bindings need to be refreshed, the user will need to switch channels (which always fetches new bindings), or refresh the page.
```

File: site/content/integrate/apps/_index.md [hunk 14]
```diff
@@ -31,6 +31,22 @@ Read the [JavaScript start guide]({{< ref "quick-start-js" >}}) or [Go quick sta
 
 ## FAQ
 
+### When would you build an app vs a custom slash command vs a webhook vs a plugin?
+
+That depends on your use case, as they each have benefits. Apps can be advertised, shared, and installed by others via the Marketplace. Comparatively, a custom slash command requires some configuration and is less interactive.
```
Comment (hanzei) [inline site/content/integrate/apps/_index.md:36]: > Apps can be advertised, shared, and installed by others via the Marketplace.

That is not the case right now.

File: site/content/integrate/apps/_index.md [hunk 15]
```diff
@@ -31,6 +31,22 @@ Read the [JavaScript start guide]({{< ref "quick-start-js" >}}) or [Go quick sta
...
+
+The built-in incoming webhook is rigid and requires specific fields to be posted to it's URL, but some external systems don't allow the customization of the fields included in the webhook payload. Compared to built-in webhooks, An App can be customized to receive an HTTP webhook posting from another system, and can process the incoming data then augment it or make an actionable message in a channel.
```
Comment (hanzei) [inline site/content/integrate/apps/_index.md:38]: 2/5 that customizable webhooks are not an outstanding feature for apps. You can do the same with a simple proxy.

File: site/content/integrate/apps/_index.md [hunk 16]
```diff
@@ -31,6 +31,22 @@ Read the [JavaScript start guide]({{< ref "quick-start-js" >}}) or [Go quick sta
...
+
+A plugin should be used when you need to alter the React UX in Mattermost or have an application that requires low-latency with the server (such as replacing characters in any message). Currently, plugins have several UX hooks that Apps cannot access, however we plan to add/migrate more UX hooks into the Apps Framework over time.
```
Comment (hanzei) [inline site/content/integrate/apps/_index.md:40]: What does "Reacht UX" mean?

**Comment by justinegeffen** [comment]
@hanzei and @mickmister: I've added your feedback/suggestions in new PRs. :)

**Pull request merged**